modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/index/ResumeCreateIndexTest.java

```diff
@@ -326,6 +339,87 @@ public void testPartialCheckpointNewIndexRows() throws Exception {
         assertEquals(100_000, selectPersonByName(n.cache(cacheName)).size());
     }
 
+    /**
+     * Checks that if there is no checkpoint after the index is created and the
+     * node is restarted, the indexes will be rebuilt.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testIncompleteIndexDroppedOnCacheDestroy() throws Exception {
+        final String cacheName = DEFAULT_CACHE_NAME;
+        final int cacheSize = 10_000;
+
+        IgniteEx n = prepareNodeToCreateNewIndex(cacheName, cacheSize, false);
+        populate(n.cache(DEFAULT_CACHE_NAME + 2), cacheSize);
+
+        IgniteEx cli = startClientGrid(1);
+
+        String idxName = "IDX0";
+        StopBuildIndexConsumer slowdownIdxCreateConsumer = new BuildIndexConsumer(getTestTimeout(), 1000);
+        addIdxCreateCacheRowConsumer(nodeName(n), idxName, slowdownIdxCreateConsumer);
+
+        IgniteInternalFuture<List<List<?>>> createIdxFut = createIdxAsync(cli.cache(cacheName), idxName);
+
+        slowdownIdxCreateConsumer.startBuildIdxFut.get(getTestTimeout());
+        checkInitStatus(n, cacheName, false, 1);
+        slowdownIdxCreateConsumer.finishBuildIdxFut.onDone();
+
+        cli.cache(DEFAULT_CACHE_NAME).destroy();
+        assertTrue(createIdxFut.isDone());
+
+        cli.createCache(cacheConfig(DEFAULT_CACHE_NAME));
+        populate(n.cache(cacheName), cacheSize);
+
+        checkCompletedStatus(n, cacheName);
+
+        StopBuildIndexConsumer stopRebuildIdxConsumer = addSlowdownIdxCreateConsumer(n, idxName, 0);
+        createIdxFut = createIdxAsync(cli.cache(cacheName), idxName);
+
+        stopRebuildIdxConsumer.startBuildIdxFut.get(getTestTimeout());
+        checkInitStatus(n, cacheName, false, 1);
+        stopRebuildIdxConsumer.finishBuildIdxFut.onDone();
+
+        createIdxFut.get(getTestTimeout());
+
+        checkCompletedStatus(n, cacheName);
+        assertTrue(allIndexes(n).containsKey(new QueryIndexKey(cacheName, idxName)));
+
+        assertEquals(cacheSize, selectPersonByName(n.cache(cacheName)).size());
+    }
+
+    /**
+     * Consumer that fails building indexes of cache.
+     */
+    static class BuildIndexConsumer extends StopBuildIndexConsumer {
```

Comment:
I think you can use:
**org.apache.ignite.internal.processors.cache.index.IndexingTestUtils.BreakBuildIndexConsumer**